.github/workflows/preview-documentation.yml

```diff
@@ -0,0 +1,44 @@
+name: preview-documentation
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build-and-deploy:
+    name: Publish documentation preview
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
+        with:
+          persist-credentials: false
+
+      - name: Build Reveal
+        working-directory: viewer
+        run: |
+          yarn
+          yarn build:prod
+
+      - name: Install and Build documentation 🔧
+        working-directory: documentation
+        run: |
+          yarn
+          yarn build
+
+      - name: Create preview folder structure
+        working-directory: documentation
+        run: |
+          export PR_NUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+          mkdir -p "docs-preview/$PR_NUMBER"
+          ls "docs-preview/"
+          cp -R build/ "docs-preview/$PR_NUMBER/"
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@releases/v3
+        with:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+          BRANCH: gh-pages # The branch the action should deploy to.
```

Comment:
I'm a bit concerned that it's the same branch we use for published docs. Also, as I understand, that action will just create folders for each PR without cleaning them up. 

I think we should look for something else.

Reply:
I agree. Branch to test concept. I think we should move doc to a separate repo too to avoid repo growing in size

Reply:
I think also we should clean up but cleanup in git isn't really saving any space since history is kept anyways